pkg/logs/sender/sender.go

```diff
@@ -46,34 +39,36 @@ func (s *Sender) Stop() {
 	<-s.done
 }
 
-// run lets the sender wire messages
+// run lets the sender send messages.
 func (s *Sender) run() {
 	defer func() {
 		s.done <- struct{}{}
 	}()
 	for payload := range s.inputChan {
-		s.wireMessage(payload)
+		s.send(payload)
 	}
 }
 
-// wireMessage lets the Sender send a message to datadog's intake
-func (s *Sender) wireMessage(payload message.Message) {
+// send keeps trying to send the message to the main destination until it succeeds
+// and try to send the message to the additional destinations only once.
+func (s *Sender) send(payload message.Message) {
 	for {
-		if s.conn == nil {
-			s.conn = s.connManager.NewConnection() // blocks until a new conn is ready
-		}
-		frame, err := s.delimiter.delimit(payload.Content())
+		err := s.destinations.Main.Write(payload)
 		if err != nil {
-			log.Error("can't send payload: ", payload, err)
-			continue
+			switch err.(type) {
+			case *FramingError:
```

Comment:
Could you explain what we're addressing with this `FramingError`? why do we want to fix it now?